libbeat/processors/add_cloud_metadata/provider_azure_vm.go

```diff
@@ -51,10 +85,147 @@ var azureVMMetadataFetcher = provider{
 				},
 				"region": c.Str("location"),
 			}.Apply(m)
-			return mapstr.M{"cloud": out}
+
+			azure, _ := s.Schema{
+				"resourcegroup": s.Object{
+					"name": c.Str("resourceGroupName"),
+				},
+			}.Apply(m)
+			return mapstr.M{"cloud": cloud, "azure": azure}
 		}
 
-		fetcher, err := newMetadataFetcher(config, "azure", azHeaders, metadataHost, azSchema, azMetadataURI)
+		azGenSchema := func(m map[string]interface{}) mapstr.M {
+			orchestrator := mapstr.M{
+				"orchestrator": mapstr.M{},
+			}
+
+			orchestrator.DeepUpdate(m)
+			return orchestrator
+		}
+
+		// hfetcher represents an http fetcher to retrieve metadata from azure metadata endpoint
+		hfetcher, err := newMetadataFetcher(config, "azure", azHeaders, metadataHost, azHttpSchema, azMetadataURI)
+		if err != nil {
+			return hfetcher, err
```

Comment:
Can we add `fmt.Errorf` statements to the errors here? We put them in some of the later functions, so we should be consistent.

Reply:
@fearful-symmetry Done